react/components/LayoutModeSwitcher.js

```diff
@@ -6,40 +6,32 @@ import Grid from '../images/Grid'
 import SingleItemGrid from '../images/SingleItemGrid'
 import InlineGrid from '../images/InlineGrid'
 
-const onChangeHandler = (type, onChange) => e => onChange(e, type)
 
 export default function LayoutModeSwitcher({ activeMode, onChange }) {
-  return (
-    <div className="vtex-layout-switcher flex justify-between pv2">
-      <div className="flex justify-center flex-auto br b--muted-4">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('small', onChange)}
-        >
-          <Grid active={activeMode === 'small'} />
-        </Button>
-      </div>
-      <div className="flex justify-center flex-auto br b--muted-4">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('inline', onChange)}
-        >
-          <InlineGrid active={activeMode === 'inline'} />
-        </Button>
-      </div>
-      <div className="flex justify-center flex-auto">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('normal', onChange)}
-        >
-          <SingleItemGrid active={activeMode === 'normal'} />
-        </Button>
+
+  const onChangeHandler = (type, onChange) => e => onChange(e, type)
```

Comment:
I think that there is no need to create this helpers functions `onChangeHandler` and `renderIcon`since they are not called more than one time in different places of the code, do not you think?

Reply:
I agree with the onChangeHandler its not usefull in our scenario, but renderIcons increase the legibility of the code in my opinion.